pwa/src/components/TestResultsForm.test.tsx

```diff
@@ -0,0 +1,195 @@
+
+    // ensure negative is unchecked and positive is checked
+    expect(negativeRadioButton).not.toHaveClass('am-radio-checked')
+    expect(positiveRadioButton).toHaveClass('am-radio-checked')
+})
+
+test('blood-sugar-results div is only displayed if blood sugar test is selected', () => {
+    const { getByTestId, queryByTestId } = render(<TestResultsForm />)
+
+    // ensure the blood-sugar-results div is not displayed
+    const re = new RegExp('^blood-sugar-results$', 'gi')
+    expect(queryByTestId(re)).toBeFalsy()
+
+    // check the blood-sugar-test checkbox
+    const checkboxItem = getByTestId('blood-sugar-test')
+    expect(checkboxItem).toHaveClass('am-checkbox-item')
+    const checkBoxInput = checkboxItem
+        .getElementsByTagName('span')[0]
+        .getElementsByClassName('am-checkbox-input')[0]
+    fireEvent.click(checkBoxInput)
+
+    // ensure the blood-sugar-results div is displayed
+    expect(getByTestId(re)).toBeTruthy()
+
+    // uncheck the checkbox
+    fireEvent.click(checkBoxInput)
+
+    // ensure the blood-sugar-results div is no longer displayed
+    expect(queryByTestId(re)).not.toBeTruthy()
+})
+
+test('blood sugar results fields is a number input box', () => {
+    const { getByTestId } = render(<TestResultsForm />)
+
+    // check the blood-sugar-test checkbox to display the blood-sugar-results div
+    const checkBoxInput = getByTestId('blood-sugar-test')
+        .getElementsByTagName('span')[0]
+        .getElementsByClassName('am-checkbox-input')[0]
+    fireEvent.click(checkBoxInput)
+
+    // ensure result field is displayed as a number input box
+    expect(getByTestId('blood-sugar-results-val')).toBeTruthy()
+    expect(getByTestId('blood-sugar-results-val')).toHaveAttribute(
+        'type',
+        'text',
+    )
+})
+
+test('ua-results div is only displayed if UA test is selected', () => {
+    const { getByTestId, queryByTestId } = render(<TestResultsForm />)
+
+    // ensure the ua-results div is not displayed
+    const re = new RegExp('^ua-results$', 'gi')
+    expect(queryByTestId(re)).toBeFalsy()
+
+    // check the ua-test checkbox
+    const checkboxItem = getByTestId('ua-test')
+    expect(checkboxItem).toHaveClass('am-checkbox-item')
+    const checkBoxInput = checkboxItem
+        .getElementsByTagName('span')[0]
+        .getElementsByClassName('am-checkbox-input')[0]
+    fireEvent.click(checkBoxInput)
+
+    // ensure the ua-results div is displayed
+    expect(getByTestId(re)).toBeTruthy()
+
+    // uncheck the checkbox
+    fireEvent.click(checkBoxInput)
+
+    // ensure the ua-results div is no longer displayed
+    expect(queryByTestId(re)).not.toBeTruthy()
+})
+
+test('ua test results are rendered as number input boxes', () => {
+    const { getByTestId, queryByTestId } = render(<TestResultsForm />)
+    const uaResultFieldID = [
+        'glucose-results-val',
+        'nitrites-results-val',
+        'protein-results-val',
+        'leukocytes-results-val',
+    ]
```

Comment:
could this list of fields be extracted out into its own data structure that both the main component and tests could share?

Reply:
done in this commit f4a2af5c04f38bfcd41a45343c6fea5ff5fb9872